packages/expo/src/dom/base.ts

```diff
@@ -0,0 +1,51 @@
+let cachedBaseUrl: string | null = null;
+
+/**
+ * Get the base URL for the DOM Components HTML
+ */
+export function getBaseURL(): string {
+  if (cachedBaseUrl != null) {
+    return cachedBaseUrl;
+  }
+
+  // Serving from updates
+  const ExpoUpdates = globalThis.expo?.modules?.['ExpoUpdates'] as
+    | import('expo-updates').ExpoUpdatesModule
+    | undefined;
+  const updatesIsInstalledAndEnabled = ExpoUpdates?.isEnabled ?? false;
+  const updatesIsEmbeddedLaunch = ExpoUpdates?.isEmbeddedLaunch ?? false;
+  const shouldServeDomFromUpdates = updatesIsInstalledAndEnabled && !updatesIsEmbeddedLaunch;
+  // If updates is installed and enabled, and we're not running from an embedded launch, we should serve the DOM Components from the `.expo-internal` directory
+  if (shouldServeDomFromUpdates) {
+    const localAssets = ExpoUpdates?.localAssets ?? {};
+    const anyLocalAsset = Object.values(localAssets)[0];
+    if (anyLocalAsset) {
+      // Try to get the `.expo-internal` directory from the first local asset
+      cachedBaseUrl = anyLocalAsset.slice(0, anyLocalAsset.lastIndexOf('/'));
+      return cachedBaseUrl;
+    }
+  }
+
+  if (process.env.EXPO_OS === 'web') {
```

Comment:
Web case shouldn't matter right? All content is just passed through.

Reply:
yes, on web, the call path should not go to `webview-wrapper.tsx -> base.ts`. the web code here currently is only used for unit test